### PR TITLE
Upgrade to to File NIO 

### PR DESCRIPTION
**Pull request by truongdatnhan**
- Update dependencies
- Upgrade to Path (File NIO) instead of File IO (Haven't been fully replaced yet)
- try-with-resource in missing place

### CODE REVIEW

File: src/main/java/client/Stat.java [hunk 1]
```diff
@@ -76,44 +76,10 @@ public static Stat getBy5ByteEncoding(int encoded) {
...
-        } else if (type.equals("FAME")) {
-            return FAME;
-        } else if (type.equals("MESO")) {
-            return MESO;
-        } else if (type.equals("PET")) {
-            return PET;
+    	for (Stat stat : Stat.values()) {
+            if (stat.name().equals(type)) {
+                return stat;
+            }
```
Comment (P0nk) [inline src/main/java/client/Stat.java:82]: Good improvement, but I can't find any usages of this function at all. Can be removed altogether.

File: src/main/java/provider/wz/XMLWZFile.java [hunk 2]
```diff
@@ -29,58 +29,64 @@
...
+                    wzdir.addDirectory(newDir);
+                    fillMapleDataEntitys(path, newDir);
+            	} else if (fileName.endsWith(".xml")) {
+            		wzdir.addFile(new WZFileEntry(fileName.substring(0, fileName.length() - 4), 0, 0, wzdir));
+            	}
             }
-        }
+        } catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```
Comment (P0nk) [inline src/main/java/provider/wz/XMLWZFile.java:62]: Should not print to standard out like this. _Instead, create a static logger field and log it as a warning.

File: src/main/java/provider/wz/XMLWZFile.java [hunk 3]
```diff
@@ -29,58 +29,64 @@
...
-    private void fillMapleDataEntitys(File lroot, WZDirectoryEntry wzdir) {
-        for (File file : lroot.listFiles()) {
-            String fileName = file.getName();
-            if (file.isDirectory() && !fileName.endsWith(".img")) {
-                WZDirectoryEntry newDir = new WZDirectoryEntry(fileName, 0, 0, wzdir);
-                wzdir.addDirectory(newDir);
-                fillMapleDataEntitys(file, newDir);
-            } else if (fileName.endsWith(".xml")) {
-                wzdir.addFile(new WZFileEntry(fileName.substring(0, fileName.length() - 4), 0, 0, wzdir));
+    private void fillMapleDataEntitys(Path lroot, WZDirectoryEntry wzdir) {
```
Comment (P0nk) [inline src/main/java/provider/wz/XMLWZFile.java:47]: Strange formatting in this method

File: src/main/java/provider/wz/XMLWZFile.java [hunk 4]
```diff
@@ -29,58 +29,64 @@
...
 
     @Override
     public synchronized Data getData(String path) {
-        File dataFile = new File(root, path + ".xml");
-        File imageDataDir = new File(root, path);
-        if (!dataFile.exists()) {
+        Path dataFile = root.resolve(path + ".xml"); 
+        //File(root, path + ".xml");
+        Path imageDataDir = root.resolve(path);
+        //File imageDataDir = new File(root, path);
```
Comment (P0nk) [inline src/main/java/provider/wz/XMLWZFile.java:71]: Get rid of the comments, they don't serve a purpose

File: src/main/java/tools/mapletools/BossHpBarFetcher.java [hunk 5]
```diff
@@ -129,21 +134,21 @@ private static void translateToken(String token) {
...
+        		if(Files.isRegularFile(path)) {
+        			try(BufferedReader br = Files.newBufferedReader(path)) {
+        				bufferedReader = br;
+        				String line;
+        				while ((line = bufferedReader.readLine()) != null) {
+                            translateToken(line);
+                        }
+        			}
+        		}
+        	}
```
Comment (P0nk) [inline src/main/java/tools/mapletools/BossHpBarFetcher.java:151]: Strange formatting. Should use 4 space indentation.

File: src/main/java/tools/mapletools/CashDropFetcher.java [hunk 6]
```diff
@@ -260,7 +262,8 @@ private static void reportNxDropResults(boolean dropdata) throws SQLException {
     }
 
     private static void reportNxDropData() {
-        try {
+        //NEED FUTURE UPDATE
```
Comment (P0nk) [inline src/main/java/tools/mapletools/CashDropFetcher.java:265]: Why does it need a future update?

File: src/main/java/tools/mapletools/CashVegaChecker.java [hunk 7]
```diff
@@ -154,11 +156,11 @@ private static void printReportFileHeader() {
     }
 
     private static void reportMissingVegaItems() {
-        System.out.println("Reporting results ...");
+        //NEED FUTURE UPDATE
```
Comment (P0nk) [inline src/main/java/tools/mapletools/CashVegaChecker.java:159]: Why does it need a future update?

File: src/main/java/tools/mapletools/DojoUpdate.java [hunk 8]
```diff
@@ -16,8 +22,10 @@
  * Estimated parse time: 10 seconds
  */
 public class DojoUpdate {
-    private static final File INPUT_DIRECTORY = new File(WZFiles.MAP.getFile(), "/Map/Map9");
-    private static final File OUTPUT_DIRECTORY = ToolConstants.getOutputFile("dojo-maps");
+    //private static final Path INPUT_DIRECTORY = WZFiles.MAP.getFile().resolve("/Map/Map9");
```
Comment (P0nk) [inline src/main/java/tools/mapletools/DojoUpdate.java:25]: Remove comment

**Review by P0nk — CHANGES_REQUESTED** [review]
Good improvements. 
There is only some clean up to do before I can merge this. I commented on formatting a few times before I realized it needs to be fixed in almost all files.
Please fix formatting (especially indentation is way off in many places) and remove unnecessary comments.

File: .gitignore [hunk 9]
```diff
@@ -1,4 +1,4 @@
-﻿/logs/**
+/logs/**
```
[the remark this answers is not in the document]
Comment (truongdatnhan) [reply .gitignore:1]: No idea why this, shows up at change. The only change I made is to add exclusion to Eclipse project file

File: handbook/Use.txt [hunk 10]
```diff
@@ -1,4 +1,4 @@
-﻿2000000 - Red Potion - A potion made out of red herbs.\nRecovers 50 HP.
+2000000 - Red Potion - A potion made out of red herbs.\nRecovers 50 HP.
```
[the remark this answers is not in the document]
Comment (truongdatnhan) [reply handbook/Use.txt:1]: There appears to be a hidden space character in the first line. I just delete it and add it again, and then it works.

File: pom.xml [hunk 11]
```diff
@@ -25,15 +25,15 @@
...
+        <log4j.version>2.18.0</log4j.version> <!-- Slf4j implementation -->
+        <graalvm.version>22.2.0</graalvm.version> <!-- ScriptEngine implementation -->
+        <netty.version>4.1.79.Final</netty.version> <!-- Networking -->
         <junit.version>5.8.2</junit.version> <!-- Unit test -->
         <yamlbeans.version>1.15</yamlbeans.version> <!-- Config file -->
         <jcip-annotations.version>1.0</jcip-annotations.version> <!-- Annotations for concurrency documentation -->
         <commons-io.version>2.11.0</commons-io.version> <!-- Util library used by some of our tools -->
         <HikariCP.version>5.0.1</HikariCP.version> <!-- Database connection pool -->
-        <mysql-connector-java.version>8.0.28</mysql-connector-java.version> <!-- MySQL JDBC driver -->
+        <mysql-connector-java.version>8.0.29</mysql-connector-java.version> <!-- MySQL JDBC driver -->
```
[the remark this answers is not in the document]
Comment (truongdatnhan) [reply pom.xml:36]: No clue about 8.0.30 got released

File: src/main/java/tools/mapletools/CashDropFetcher.java (hunk 6 above)
Comment (truongdatnhan) [reply src/main/java/tools/mapletools/CashDropFetcher.java:265]: I haven't implements Java NIO on this file

**Comment by truongdatnhan** [comment]
I did a new pull request removing comments and upgrading to Java NIO in some remaining files. Also fixing formatting for each!

**Comment by P0nk** [comment]
Indentation is still off. Either configure Eclipse (or whatever IDE using) to indent with 4 spaces, or switch to IntelliJ where that's the default.

**Comment by truongdatnhan** [comment]
> 

I think I have fixed it; check again.

**Review by P0nk — APPROVED** [review]

**Pull request merged**